### PR TITLE
Fixes #162 - Load external resources

### LINKED ISSUE #162: Dynamically configure maven dependencies for lightblue plugins
From a maven dependency perspective, all possible plugins must be listed in lightblue-rest's pom file. You can already fine tune which plugins are actually used at runtime via the lightblue-metadata.json configuration, but all the possible plugins must still be listed in the pom.

It would be idea if you could truly configure what plugins you wanted to use. This would be beneficial for highly customized plugins that not anyone would want to use, or if you just want a lighter weight installation.

### PR DESCRIPTION
**Pull request by dcrissman**
Allows lightblue-rest to load external jar files and directories. It does this by adding a URLClassLoader to the currentThread's class loader.

I also removed the ldap hard requirement as a test.

This is a part of #162.

### CODE REVIEW

**Comment by dcrissman** [comment]
Ready for review

**Comment by dcrissman** [comment]
This possibly constitutes a breaking change as it removes ldap as a hard dependency. Not a big deal if a deployment doesn't use ldap, but if it does, then ldap will now need to be deployed separately.

File: common/src/main/java/com/redhat/lightblue/rest/ExternalResourceConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+    private final Set<URL> externalPaths = new HashSet<>();
+
+    public Set<URL> getExternalUrls() {
+        return Collections.unmodifiableSet(externalPaths);
+    }
+
+    public ExternalResourceConfiguration() {}
+
+    public ExternalResourceConfiguration(JsonNode node) throws IOException {
```
Comment (BVulaj) [inline common/src/main/java/com/redhat/lightblue/rest/ExternalResourceConfiguration.java]: What's the point of initializing this from a jsonnode?  Maybe some documentation around this if we expect to read in a json file or something.

File: common/src/main/java/com/redhat/lightblue/rest/ExternalResourceConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        if (!file.exists()) {
+            throw new IllegalArgumentException("File does not exist: " + file.getAbsolutePath());
+        }
+
+        if (file.isDirectory()) {
+            File[] files = file.listFiles(new FilenameFilter() {
+
+                @Override
+                public boolean accept(File dir, String name) {
+                    if (name.endsWith(".jar")
```
Comment (BVulaj) [inline common/src/main/java/com/redhat/lightblue/rest/ExternalResourceConfiguration.java]: Given this, won't externalPaths end up with directories as well as jar flies?  Shouldn't this recurse and find them all?

File: common/src/main/java/com/redhat/lightblue/rest/ExternalResourceConfiguration.java (hunk 1 above)
Comment (dcrissman) [inline common/src/main/java/com/redhat/lightblue/rest/ExternalResourceConfiguration.java]: The class level javadoc includes an example of what the json config should look like.

**Pull request merged**